pennylane_lightning/core/src/simulators/lightning_tensor/utils/tncuda_utils/tncuda_helpers.hpp

```diff
@@ -194,4 +194,21 @@ inline auto create_swap_wire_pair_queue(const std::vector<std::size_t> &wires)
     return {local_wires, swap_wires_queue};
 }
 
+/**
+ * @brief Check if the provided MPS has the correct dimension for C++
+ * backend.
+ *
+ * @param MPS_shape_dest Dimension list of destination MPS.
+ * @param MPS_shape_source Dimension list of incoming MPS.
+ */
+inline void
+MPSShapeCheck(const std::vector<std::vector<std::size_t>> &MPS_shape_dest,
+              const std::vector<std::vector<std::size_t>> &MPS_shape_source) {
+    if (MPS_shape_dest != MPS_shape_source) {
+        PL_ABORT_IF_NOT(MPS_shape_dest == MPS_shape_source,
```

Comment:
and in fact, if this `MPSShapeCheck` is just a `PL_ABORT_IF_NOT` , maybe we can just replace all instance of `MPSShapeCheck` to the `PL_ABORT_IF_NOT`  in `bindings`? it's only used once and it's quite short

Reply:
I prefer to keep this function here due to Shuli's [suggestion](https://github.com/PennyLaneAI/pennylane-lightning/pull/983#discussion_r1888589332).